### PR TITLE
PPTP-1420 : Post Registration - Add New Group Member

### PR DESCRIPTION
**Pull request by RyanKellyHMRC**
Support newly added group members when transforming registrations to PPT subscriptions.

#### Check list 
 - [x] `./precheck` was executed (Integration/Component/Unit tests)
 - [x] `sbt scalafmt test:scalafmt` was executed

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/models/eis/subscription/group/GroupPartnershipSubscription.scala [hunk 1]
```diff
@@ -134,10 +148,13 @@ object GroupPartnershipSubscription {
                             addressDetails =
                               AddressDetails(organisationDetails.registeredBusinessAddress),
                             contactDetails = ContactDetails(primaryContactDetails),
-                            regWithoutIDFlag = organisationDetails.regWithoutIDFlag
+                            regWithoutIDFlag =
+                              if (isUpdate)
+                                Some(organisationDetails.regWithoutIDFlag.getOrElse(false))
+                              else organisationDetails.regWithoutIDFlag
```
Comment (tonymccrae-ee) [inline app/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/models/eis/subscription/group/GroupPartnershipSubscription.scala:154]: Could we have a 1 liner explain the motive behind this change?

File: app/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/models/eis/subscription/group/GroupPartnershipSubscription.scala [hunk 2]
```diff
@@ -117,12 +127,16 @@ object GroupPartnershipSubscription {
                               partner.contactDetails.getOrElse(
                                 throw new IllegalStateException("Partner contact details absent")
                               )
-                            )
+                            ),
+                            regWithoutIDFlag =
+                              if (isUpdate) Some(false)
+                              else None // TODO: should this flag be added to Partner
```
Comment (paulblakehmrc) [inline app/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/models/eis/subscription/group/GroupPartnershipSubscription.scala:133]: Required for partner, same rules as group members

**Review by paulblakehmrc — APPROVED** [review]
one comment on a todo

File: app/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/models/eis/subscription/group/GroupPartnershipSubscription.scala (hunk 2 above)
Comment (RyanKellyHMRC) [reply app/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/models/eis/subscription/group/GroupPartnershipSubscription.scala]: I have modified the TODO message to make it clearer that this was about a reminder to add a `regWithoutIDFlag` to the `Partner` class in the future when we get around to building the post registration amendment facility for partner registrations.

File: app/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/models/eis/subscription/group/GroupPartnershipSubscription.scala (hunk 1 above)
Comment (RyanKellyHMRC) [reply app/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/models/eis/subscription/group/GroupPartnershipSubscription.scala:161]: I've refactored a little to help to make the intention clearer and provided a more detailed commit comment.

**Pull request merged**